app/models/concerns/operations/stock_operations.rb

```diff
@@ -68,5 +70,98 @@ def register_loss(package, quantity:, from_location:)
         location: from_location
       )
     end
+
+    def pack_or_unpack(params, user_id)
+      raise ActionNotAllowedError.new unless PACK_UNPACK_ALLOWED_ACTIONS.include?(params[:task])
+      PackUnpack.new(params, user_id).public_send(params[:task])
+    end
+
+    class PackUnpack
+      def initialize(params, user_id)
+        @cause = Package.find(params[:id]) # box or pallet
+        @item = Package.find(params[:item_id]) # item to add or remove
```

Comment:
could we use the term `package` instead of `item`. Item has a meaning in the api, it is a separate model

Reply:
sure @patrixr .